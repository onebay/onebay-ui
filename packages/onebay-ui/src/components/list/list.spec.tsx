import { mount } from '@vue/test-utils'
import List from './index'
import ListItem from './item/index'

const factory = (values = {}, slots = { default: [] }) => {
  return mount(List, {
    components: {
      ListItem,
    },
    slots,
    props: { ...values }
  })
}

describe('List Snap', () => {
  it('render completed List', () => {
    const wrapper = mount({
      data() {
        return {
          iconInfo: {
            size: 25,
            color: '#78A4FA',
            value: 'calendar',
          },
        }
      },
      components: {
        List,
        ListItem,
      },
      template: `
      <List>
        <ListItem title='Title text' />
        <ListItem title='Title text' arrow='right' />
        <ListItem title='Title text' note='Description' />
        <ListItem title='Disabled state' disabled extraText='details' />
        <ListItem title='Title text' note='Description' arrow='right' />
        <ListItem title='Title text' extraText='details' arrow='right' />
        <ListItem
          arrow='right'
          note='Description'
          title='Title textTitle textTitle textTitle textTitle text'
          extraText='details details details details'
        />
        <ListItem
          title='Title text'
          note='Description'
          extraText='details'
          arrow='right'
          thumb='http://img12.360buyimg.com/jdphoto/s72x72_jfs/t10660/330/203667368/1672/801735d7/59c85643N31e68303.png'
        />
        <ListItem
          arrow='right'
          note='Description'
          :iconInfo="iconInfo"
          title='Title text'
          extraText='details'
          thumb='http://img12.360buyimg.com/jdphoto/s72x72_jfs/t10660/330/203667368/1672/801735d7/59c85643N31e68303.png'
        />
        <ListItem title='Title text' isSwitch />
        <ListItem title='Title text' isSwitch disabled />
        <ListItem title='Title text' switchIsCheck isSwitch />
        <ListItem title='Title text' switchIsCheck isSwitch disabled />
      </List>
      `,
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render completed List -- no border', () => {
    const wrapper = mount({
      components: {
        List,
        ListItem,
      },
      template: `
      <List :hasBorder="false">
        <ListItem title='Title text' :hasBorder="false" />
        <ListItem title='Title text' :hasBorder="false" />
      </List>
      `,
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})

describe('List Behavior ', () => {
  it('ListItem onClick', () => {
    const onClick = jest.fn()
    const wrapper = mount(
      <List>
        <ListItem title='标题文字' onClick={onClick} />
      </List>, {
      props: {
        title: 'ListItem title'
      }
    })
    wrapper.find('.ob-list .ob-list__item').trigger('click')
    expect(wrapper.findComponent(ListItem).emitted()).toHaveProperty('click')
  })
})
