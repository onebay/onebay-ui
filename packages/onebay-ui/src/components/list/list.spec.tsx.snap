// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Snap render completed List -- no border 1`] = `
<div
  class="ob-list ob-list--no-border"
>
  
  <div
    class="ob-list__item ob-list__item--no-border"
  >
    <div
      class="ob-list__item-container"
    >
      
      
      <div
        class="ob-list__item-content item-content"
      >
        <div
          class="item-content__info"
        >
          <div
            class="item-content__info-title"
          >
            Title text
          </div>
          
        </div>
      </div>
      <div
        class="ob-list__item-extra item-extra"
      >
        
        
        
      </div>
    </div>
  </div>
  <div
    class="ob-list__item ob-list__item--no-border"
  >
    <div
      class="ob-list__item-container"
    >
      
      
      <div
        class="ob-list__item-content item-content"
      >
        <div
          class="item-content__info"
        >
          <div
            class="item-content__info-title"
          >
            Title text
          </div>
          
        </div>
      </div>
      <div
        class="ob-list__item-extra item-extra"
      >
        
        
        
      </div>
    </div>
  </div>
  
</div>
`;

exports[`List Snap render completed List 1`] = `
<div
  class="ob-list"
>
  
  <div
    class="ob-list__item"
  >
    <div
      class="ob-list__item-container"
    >
      
      
      <div
        class="ob-list__item-content item-content"
      >
        <div
          class="item-content__info"
        >
          <div
            class="item-content__info-title"
          >
            Title text
          </div>
          
        </div>
      </div>
      <div
        class="ob-list__item-extra item-extra"
      >
        
        
        
      </div>
    </div>
  </div>
  <div
    class="ob-list__item"
  >
    <div
      class="ob-list__item-container"
    >
      
      
      <div
        class="ob-list__item-content item-content"
      >
        <div
          class="item-content__info"
        >
          <div
            class="item-content__info-title"
          >
            Title text
          </div>
          
        </div>
      </div>
      <div
        class="ob-list__item-extra item-extra"
      >
        
        
        <div
          class="item-extra__icon"
        >
          <div
            class="ob-icon item-extra__icon-arrow ob-icon-chevron-right"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="ob-list__item ob-list__item--multiple"
  >
    <div
      class="ob-list__item-container"
    >
      
      
      <div
        class="ob-list__item-content item-content"
      >
        <div
          class="item-content__info"
        >
          <div
            class="item-content__info-title"
          >
            Title text
          </div>
          <div
            class="item-content__info-note"
          >
            Description
          </div>
        </div>
      </div>
      <div
        class="ob-list__item-extra item-extra"
      >
        
        
        
      </div>
    </div>
  </div>
  <div
    class="ob-list__item ob-list__item--disabled"
  >
    <div
      class="ob-list__item-container"
    >
      
      
      <div
        class="ob-list__item-content item-content"
      >
        <div
          class="item-content__info"
        >
          <div
            class="item-content__info-title"
          >
            Disabled state
          </div>
          
        </div>
      </div>
      <div
        class="ob-list__item-extra item-extra"
      >
        <div
          class="item-extra__info"
        >
          details
        </div>
        
        
      </div>
    </div>
  </div>
  <div
    class="ob-list__item ob-list__item--multiple"
  >
    <div
      class="ob-list__item-container"
    >
      
      
      <div
        class="ob-list__item-content item-content"
      >
        <div
          class="item-content__info"
        >
          <div
            class="item-content__info-title"
          >
            Title text
          </div>
          <div
            class="item-content__info-note"
          >
            Description
          </div>
        </div>
      </div>
      <div
        class="ob-list__item-extra item-extra"
      >
        
        
        <div
          class="item-extra__icon"
        >
          <div
            class="ob-icon item-extra__icon-arrow ob-icon-chevron-right"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="ob-list__item"
  >
    <div
      class="ob-list__item-container"
    >
      
      
      <div
        class="ob-list__item-content item-content"
      >
        <div
          class="item-content__info"
        >
          <div
            class="item-content__info-title"
          >
            Title text
          </div>
          
        </div>
      </div>
      <div
        class="ob-list__item-extra item-extra"
      >
        <div
          class="item-extra__info"
        >
          details
        </div>
        
        <div
          class="item-extra__icon"
        >
          <div
            class="ob-icon item-extra__icon-arrow ob-icon-chevron-right"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="ob-list__item ob-list__item--multiple"
  >
    <div
      class="ob-list__item-container"
    >
      
      
      <div
        class="ob-list__item-content item-content"
      >
        <div
          class="item-content__info"
        >
          <div
            class="item-content__info-title"
          >
            Title textTitle textTitle textTitle textTitle text
          </div>
          <div
            class="item-content__info-note"
          >
            Description
          </div>
        </div>
      </div>
      <div
        class="ob-list__item-extra item-extra"
      >
        <div
          class="item-extra__info"
        >
          details details details details
        </div>
        
        <div
          class="item-extra__icon"
        >
          <div
            class="ob-icon item-extra__icon-arrow ob-icon-chevron-right"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="ob-list__item ob-list__item--thumb ob-list__item--multiple"
  >
    <div
      class="ob-list__item-container"
    >
      <div
        class="ob-list__item-thumb item-thumb"
      >
        <img
          class="item-thumb__info"
          src="http://img12.360buyimg.com/jdphoto/s72x72_jfs/t10660/330/203667368/1672/801735d7/59c85643N31e68303.png"
        />
      </div>
      
      <div
        class="ob-list__item-content item-content"
      >
        <div
          class="item-content__info"
        >
          <div
            class="item-content__info-title"
          >
            Title text
          </div>
          <div
            class="item-content__info-note"
          >
            Description
          </div>
        </div>
      </div>
      <div
        class="ob-list__item-extra item-extra"
      >
        <div
          class="item-extra__info"
        >
          details
        </div>
        
        <div
          class="item-extra__icon"
        >
          <div
            class="ob-icon item-extra__icon-arrow ob-icon-chevron-right"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="ob-list__item ob-list__item--thumb ob-list__item--multiple"
  >
    <div
      class="ob-list__item-container"
    >
      <div
        class="ob-list__item-thumb item-thumb"
      >
        <img
          class="item-thumb__info"
          src="http://img12.360buyimg.com/jdphoto/s72x72_jfs/t10660/330/203667368/1672/801735d7/59c85643N31e68303.png"
        />
      </div>
      <div
        class="ob-list__item-icon item-icon"
      >
        <div
          class="ob-icon ob-icon-calendar"
          style="color: rgb(120, 164, 250); font-size: 25px;"
        />
      </div>
      <div
        class="ob-list__item-content item-content"
      >
        <div
          class="item-content__info"
        >
          <div
            class="item-content__info-title"
          >
            Title text
          </div>
          <div
            class="item-content__info-note"
          >
            Description
          </div>
        </div>
      </div>
      <div
        class="ob-list__item-extra item-extra"
      >
        <div
          class="item-extra__info"
        >
          details
        </div>
        
        <div
          class="item-extra__icon"
        >
          <div
            class="ob-icon item-extra__icon-arrow ob-icon-chevron-right"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="ob-list__item"
    isswitch=""
  >
    <div
      class="ob-list__item-container"
    >
      
      
      <div
        class="ob-list__item-content item-content"
      >
        <div
          class="item-content__info"
        >
          <div
            class="item-content__info-title"
          >
            Title text
          </div>
          
        </div>
      </div>
      <div
        class="ob-list__item-extra item-extra"
      >
        
        
        
      </div>
    </div>
  </div>
  <div
    class="ob-list__item ob-list__item--disabled"
    isswitch=""
  >
    <div
      class="ob-list__item-container"
    >
      
      
      <div
        class="ob-list__item-content item-content"
      >
        <div
          class="item-content__info"
        >
          <div
            class="item-content__info-title"
          >
            Title text
          </div>
          
        </div>
      </div>
      <div
        class="ob-list__item-extra item-extra"
      >
        
        
        
      </div>
    </div>
  </div>
  <div
    class="ob-list__item"
    isswitch=""
    switchischeck=""
  >
    <div
      class="ob-list__item-container"
    >
      
      
      <div
        class="ob-list__item-content item-content"
      >
        <div
          class="item-content__info"
        >
          <div
            class="item-content__info-title"
          >
            Title text
          </div>
          
        </div>
      </div>
      <div
        class="ob-list__item-extra item-extra"
      >
        
        
        
      </div>
    </div>
  </div>
  <div
    class="ob-list__item ob-list__item--disabled"
    isswitch=""
    switchischeck=""
  >
    <div
      class="ob-list__item-container"
    >
      
      
      <div
        class="ob-list__item-content item-content"
      >
        <div
          class="item-content__info"
        >
          <div
            class="item-content__info-title"
          >
            Title text
          </div>
          
        </div>
      </div>
      <div
        class="ob-list__item-extra item-extra"
      >
        
        
        
      </div>
    </div>
  </div>
  
</div>
`;
